pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/AbstractMetadataStore.java

```diff
@@ -117,17 +118,19 @@ public CompletableFuture<List<String>> asyncReload(String key, List<String> oldV
                 .buildAsync(new AsyncCacheLoader<String, Boolean>() {
                     @Override
                     public CompletableFuture<Boolean> asyncLoad(String key, Executor executor) {
-                        return existsFromStore(key);
+                        // force jump to forkJoinThread to avoid callback execute on the metadataThread
+                        return existsFromStore(key).thenApplyAsync(Function.identity());
                     }
 
                     @Override
                     public CompletableFuture<Boolean> asyncReload(String key, Boolean oldValue,
                             Executor executor) {
                         if (isConnected) {
-                            return existsFromStore(key);
+                            // force jump to forkJoinThread to avoid callback execute on the metadataThread
+                            return existsFromStore(key).thenApplyAsync(Function.identity());
                         } else {
                             // Do not refresh if we're not connected
-                            return CompletableFuture.completedFuture(oldValue);
+                            return CompletableFuture.supplyAsync(() -> oldValue);
```

Comment:
It's better to use the `executor` in this class.

Reply:
hi @Technoboy- after read the Caffeine `asyncReload` method javadoc. i change back to the origin ·CompletableFuture.completedFuture` because the `asyncReload` method will be trigger on fork-join-pool which is safe to complete.